### PR TITLE
[libvpx] Update to latest and add support for arm64 on macOS

### PR DESCRIPTION
**Pull request by vejmartin**
Updates libvpx to latest and adds support for arm64 on macOS

- #### What does your PR fix?  
  N/A

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  arm64-osx, arm64-osx-dynamic, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/libvpx/0002-Fix-nasm-debug-format-flag.patch [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+           EXE_SFX=.exe
+           ;;
+         win64)
+           add_asflags -f win64
+-          enabled debug && add_asflags -g cv8
++          enabled debug && [ "${AS}" = yasm ] && add_asflags -g cv8
++          enabled debug && [ "${AS}" = nasm ] && add_asflags -gcv8
+           EXE_SFX=.exe
+           ;;
+         linux*|solaris*|android*)
```
Comment (NancyLi1013) [inline ports/libvpx/0002-Fix-nasm-debug-format-flag.patch:21]: Please keep the same format of EOL as before in this patch file. Maybe LF or CRLF.

File: ports/libvpx/0004-remove-library-suffixes.patch [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
++++ b/build/make/gen_msvs_vcxproj.sh
+@@ -394,7 +394,7 @@ generate_vcxproj() {
+               else
+                 config_suffix=""
+               fi
+-              tag_content TargetName "${name}${lib_sfx}${config_suffix}"
++              tag_content TargetName "${name}"
+             fi
+             close_tag PropertyGroup
+         done
```
Comment (NancyLi1013) [inline ports/libvpx/0004-remove-library-suffixes.patch:13]: The same as above.

File: ports/libvpx/portfile.cmake [hunk 3]
```diff
@@ -165,7 +164,11 @@ else()
 	elseif(VCPKG_TARGET_IS_LINUX)
         set(LIBVPX_TARGET "${LIBVPX_TARGET_ARCH}-linux-gcc")
     elseif(VCPKG_TARGET_IS_OSX)
-        set(LIBVPX_TARGET "${LIBVPX_TARGET_ARCH}-darwin17-gcc") # enable latest CPU instructions for best performance and less CPU usage on MacOS
+        if(VCPKG_TARGET_ARCHITECTURE STREQUAL arm64)
+            set(LIBVPX_TARGET "arm64-darwin20-gcc")
```
Comment (NancyLi1013) [inline ports/libvpx/portfile.cmake:168]: Why do you use `drawin20 `instead of `darwin17 `here?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/libvpx/portfile.cmake (hunk 3 above)
Comment (vejmartin) [reply ports/libvpx/portfile.cmake:168]: Originally the arm64 triplet was used for iOS, however, with Big Sur, it now also applies to macOS. `arm64-darwin20-gcc` is arm64 macOS and `arm64-darwin-gcc` is arm64 iOS. There's no `darwin17` support for arm64 macs.
See: https://github.com/webmproject/libvpx/commit/979e27c9700448bded96fb1d4ba31441a8da82db

File: ports/libvpx/0002-Fix-nasm-debug-format-flag.patch (hunk 1 above)
Comment (vejmartin) [reply ports/libvpx/0002-Fix-nasm-debug-format-flag.patch:21]: Thank you for catching that, I'll update it.

File: ports/libvpx/portfile.cmake (hunk 3 above)
Comment (NancyLi1013) [reply ports/libvpx/portfile.cmake:168]: Thanks for your clarification.

**Review by NancyLi1013 — APPROVED** [review]

File: ports/libvpx/portfile.cmake [hunk 4]
```diff
@@ -165,7 +164,11 @@ else()
 	elseif(VCPKG_TARGET_IS_LINUX)
         set(LIBVPX_TARGET "${LIBVPX_TARGET_ARCH}-linux-gcc")
     elseif(VCPKG_TARGET_IS_OSX)
-        set(LIBVPX_TARGET "${LIBVPX_TARGET_ARCH}-darwin17-gcc") # enable latest CPU instructions for best performance and less CPU usage on MacOS
+        if(VCPKG_TARGET_ARCHITECTURE STREQUAL arm64)
```
Comment (strega-nil-ms) [inline ports/libvpx/portfile.cmake:167]:
```suggestion
        if(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
```
the quotes are necessary.

Comment (strega-nil-ms) [reply ports/libvpx/portfile.cmake:167]: this is also really suspect given the above setting of LIBBPX_TARGET_ARCH

Comment (vejmartin) [reply ports/libvpx/portfile.cmake:167]: @strega-nil-ms, thank you, I just learned about this policy, https://cmake.org/cmake/help/latest/policy/CMP0054.html#policy:CMP0054 :)

**Comment by NancyLi1013** [comment]
Please run `vcpkg x-add-version --overwrite-version libvpx` to solve the failures on x86-windows.

File: ports/libvpx/portfile.cmake (hunk 4 above)
Comment (NancyLi1013) [reply ports/libvpx/portfile.cmake]: There are also some other places that have the same problem. Might also require quotes.

**Comment by NancyLi1013** [comment]
LGTM, except for the comments here https://github.com/microsoft/vcpkg/pull/18183#discussion_r644447947. 
@strega-nil-ms please help confirm if it's necessary to update.

Thanks.

**Pull request merged**